### ISSUE TITLE
No compiler found in PATH

### ISSUE THREAD

**Issue by sgf** [d-language-server/vscode-dlang#10]
when im enable this plugs its always "No compiler found in PATH"

but im sure the PATH is exists the DMD path

and  idont know how to use it.
what should i do?

**Comment by LaurentTreguier** [comment]
Can you do `where dub` and `where dmd` in the command console ?

**Comment by LaurentTreguier** [comment]
You can also try to do `dub fetch dls` and `dub run dls:bootstrap` in the command console, and restart VSCode.

**Comment by sgf** [comment]
```

C:\WINDOWS\system32>where dub
D:\AppDev\D\dmd2\windows\bin\dub.exe

C:\WINDOWS\system32>where dmd
D:\AppDev\D\dmd2\windows\bin\dmd.exe

C:\WINDOWS\system32>dub fetch dls
Fetching dls 0.24.4...
Please note that you need to use `dub run <pkgname>` or add it to dependencies of your package to actually use/run it. dub does not do actual installation of packages outside of its own ecosystem.

C:\WINDOWS\system32>dub run dls:bootstrap
Building package dls:bootstrap in C:\Users\super\AppData\Local\dub\packages\dls-0.24.4\dls\bootstrap\
Running pre-generate commands for dls:i18n...
Performing "debug" build using D:\AppDev\D\dmd2\windows\bin\dmd.exe for x86.
dls:protocol 0.24.4: building configuration "library"...
dls:i18n 0.24.4: building configuration "library"...
dls:bootstrap 0.24.4: building configuration "application"...
Linking...
Running ..\..\Users\super\AppData\Local\dub\packages\dls-0.24.4\dls\bootstrap\dls_bootstrap.exe

```

**Comment by LaurentTreguier** [comment]
After some time, it should stop and output `C:\Users\super\AppData\Local\dub\packages\.bin\dls-latest\dls.exe`.
Does it never stop?

**Comment by sgf** [comment]
I have this file
`C:\Users\super\AppData\Local\dub\packages\.bin\dls-v0.24.4\dls.exe`

**Comment by LaurentTreguier** [comment]
If you have this file, then now the extension should work without saying 'No compiler found in PATH"

**Comment by sgf** [comment]
thank you i will test that later.
dls.exe is new file in my disk.

**Comment by sgf** [comment]
Hi,was DLS support IntelliSense now ?

**Comment by LaurentTreguier** [comment]
If you don't have an error, DLS should support intellisense

**Comment by sgf** [comment]
have little ,but no such as parmeters info of functions .
and input chars no full functions list out.
example:
im input 'wr' it wont list out all 'wr' begin's function
of all namespace at current d file what im imports;

**Comment by LaurentTreguier** [comment]
This is partly because D is hard to parse. DLS uses [DCD](https://github.com/dlang-community/DCD) for completion.
Do you use Dub for your projects ?

**Comment by sgf** [comment]
yes,im use dub create project and build.
but just do that things.no others.
thank you.
Now I know why the D language so many years is not a easy to use IDE.
